bin/install-wp-tests.sh

```diff
@@ -107,75 +138,79 @@ install_test_suite() {
 		sed $ioption "s/yourusernamehere/$DB_USER/" "$WP_TESTS_DIR"/wp-tests-config.php
 		sed $ioption "s/yourpasswordhere/$DB_PASS/" "$WP_TESTS_DIR"/wp-tests-config.php
 		sed $ioption "s|localhost|${DB_HOST}|" "$WP_TESTS_DIR"/wp-tests-config.php
 	fi
-
 }
 
-recreate_db() {
-	shopt -s nocasematch
-	if [[ $1 =~ ^(y|yes)$ ]]
-	then
-		mysqladmin drop $DB_NAME -f --user="$DB_USER" --password="$DB_PASS"$EXTRA
-		create_db
-		echo "Recreated the database ($DB_NAME)."
-	else
-		echo "Leaving the existing database ($DB_NAME) in place."
+install_db() {
+	if [ ${SKIP_DB_CREATE} = "true" ]; then
+		return 0
 	fi
-	shopt -u nocasematch
-}
 
-create_db() {
-	mysqladmin create $DB_NAME --user="$DB_USER" --password="$DB_PASS"$EXTRA
+	wait_db
+	local MYSQLADMIN_FLAGS=$(get_db_connection_flags)
+
+	# drop database if exists
+	set +e
+	mysqladmin drop --force $DB_NAME $MYSQLADMIN_FLAGS &> /dev/null
+	set -e
+
+	# create database
+	mysqladmin create $DB_NAME $MYSQLADMIN_FLAGS
 }
 
-install_db() {
+install_woocommerce() {
+	WC_INSTALL_EXTRA=''
+	INSTALLED_WC_VERSION=$(wp plugin get woocommerce --field=version)
 
-	if [ ${SKIP_DB_CREATE} = "true" ]; then
-		return 0
+	if [[ $WC_VERSION == 'beta' ]]; then
+		# Get the latest non-trunk version number from the .org repo. This will usually be the latest release, beta, or rc.
+		WC_VERSION=$(curl https://api.wordpress.org/plugins/info/1.0/woocommerce.json | jq -r '.versions | with_entries(select(.key|match("beta";"i"))) | keys[-1]' --sort-keys)
 	fi
 
-	# parse DB_HOST for port or socket references
-	local PARTS=(${DB_HOST//\:/ })
-	local DB_HOSTNAME=${PARTS[0]};
-	local DB_SOCK_OR_PORT=${PARTS[1]};
-	local EXTRA=""
-
-	if ! [ -z $DB_HOSTNAME ] ; then
-		if [ $(echo $DB_SOCK_OR_PORT | grep -e '^[0-9]\{1,\}$') ]; then
-			EXTRA=" --host=$DB_HOSTNAME --port=$DB_SOCK_OR_PORT --protocol=tcp"
-		elif ! [ -z $DB_SOCK_OR_PORT ] ; then
-			EXTRA=" --socket=$DB_SOCK_OR_PORT"
-		elif ! [ -z $DB_HOSTNAME ] ; then
-			EXTRA=" --host=$DB_HOSTNAME --protocol=tcp"
+	if [[ -n $INSTALLED_WC_VERSION ]] && [[ $WC_VERSION == 'latest' ]]; then
+		# WooCommerce is already installed, we just must update it to the latest stable version
+		wp plugin update woocommerce
+		wp plugin activate woocommerce
+	else
+		if [[ $INSTALLED_WC_VERSION != $WC_VERSION ]]; then
+			# WooCommerce is installed but it's the wrong version, overwrite the installed version
+			WC_INSTALL_EXTRA+=" --force"
 		fi
+		if [[ $WC_VERSION != 'latest' ]] && [[ $WC_VERSION != 'beta' ]]; then
+			WC_INSTALL_EXTRA+=" --version=$WC_VERSION"
+		fi
+		wp plugin install woocommerce --activate$WC_INSTALL_EXTRA
 	fi
+}
 
-	# create database
-	if [ $(mysql --user="$DB_USER" --password="$DB_PASS"$EXTRA --execute='show databases;' | grep ^$DB_NAME$) ]
-	then
-		echo "Reinstalling will delete the existing test database ($DB_NAME)"
-		read -p 'Are you sure you want to proceed? [y/N]: ' DELETE_EXISTING_DB
-		recreate_db $DELETE_EXISTING_DB
+install_gutenberg() {
```

Comment:
why do we need to install the gutenberg plugin? isn't gutenberg in the wp core already?

Reply:
If just left it in there because Automaticc did it that way and it worked, but I'll remove it.